### PR TITLE
Add pki manager

### PR DESCRIPTION
**Pull request by hectorhuertas**

### CODE REVIEW

File: example/vault-client-namespace-role-binding.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  name: vault-namespace-admin
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: admin
+subjects:
+  - kind: ServiceAccount
+    name: vault-pki
+    # Placeholder, patch with the client namespace value
+    namespace: example
```
Comment (george-angel) [inline example/vault-client-namespace-role-binding.yaml]: You only need to specify this if  the pki lives in a different namespace to vault?

File: base/vault-pki.yaml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                  # Update secrets
+                  echo "Updating secret in ${VAULT_NAMESPACE}"
+                  kubectl -n "${VAULT_NAMESPACE}" delete secret "${secret_name}"
+                  kubectl -n "${VAULT_NAMESPACE}" create secret \
+                    generic "${secret_name}" \
+                    --from-file "${output_dir}"/ca.crt \
+                    --from-file "${output_dir}"/tls.crt \
+                    --from-file "${output_dir}"/tls.key
+
+                  for n in ${VAULT_CLIENT_NAMESPACES}; do
```
Comment (george-angel) [inline base/vault-pki.yaml]: How do you pass this in as an array?

File: base/vault-pki.yaml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+                # Validation and defaults
+                : ${VAULT_NAMESPACE:?"Need to set VAULT_NAMESPACE"}
+                : ${VAULT_CLIENT_NAMESPACES:?"Need to set VAULT_CLIENT_NAMESPACES"}
+                kubectl_version="${KUBECTL_VERSION:-"$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)"}"
+                secret_name="${VAULT_SECRET_NAME:-"vault-tls"}"
+                vault_name="${VAULT_NAME:-"vault"}"
+                replicas="${VAULT_REPLICAS:-"3"}"
+
+                if ! which curl > /dev/null; then
```
Comment (alkar) [inline base/vault-pki.yaml]: `curl` is used in setting `kubectl_version`, I think this check should be first?

File: example/vault-client-namespace-role-binding.yaml (hunk 1 above)
Comment (hectorhuertas) [reply example/vault-client-namespace-role-binding.yaml]: no, that is needed in the namespaces where clients live, so vault-pki can create the CA secret (or potentially configmap)

File: base/vault-pki.yaml (hunk 2 above)
Comment (hectorhuertas) [reply base/vault-pki.yaml]: spaced separated list

File: base/vault-pki.yaml (hunk 3 above)
Comment (hectorhuertas) [reply base/vault-pki.yaml]: good point

File: base/vault-pki.yaml (hunk 2 above)
Comment (george-angel) [reply base/vault-pki.yaml]: More things I learned

**Pull request merged**